### PR TITLE
RMMIS-6516: Support file API in UIC 3

### PR DESCRIPTION
**Pull request by kjschulz**
RMMIS-6516: Changes to File & EntryList to support CRUD and the file API in UIC 3

### CODE REVIEW

**Comment by traviscooperscdhhs** [comment]
(ride)(ride)

File: src/EntryList.jsx [hunk 1]
```diff
@@ -60,6 +60,13 @@ class EntryList extends React.Component {
         entryIndex
       }
     };
+    let entry = this.props.value[entryIndex];
+    if (this.props.containsFileFields) {
+      e.component.file = {
+        containsFileFields: this.props.containsFileFields,
```
Comment (mallenscdhhs) [inline src/EntryList.jsx]: not sure why you have this property here? it seems like you are using two checks to see if there are file edits. Why not just always send `e.component.file` and have the check in PE be for `e.component.file.containsFileFields`?

Comment (kjschulz) [inline src/EntryList.jsx]: I think you're right...will fix

Comment (kjschulz) [inline src/EntryList.jsx]: Updated

**Comment by mallenscdhhs** [comment]
@kjschulz tests are failing

**Comment by kjschulz** [comment]
(doh)

**Comment by kjschulz** [comment]
Fixed!

**Pull request merged**